### DISCUSSION TITLE
Error! Unable to generate Contract ByteCode and ABI when verifying simple storage contract on sepolia etherscan

### DISCUSSION THREAD

**Discussion by Raghav-Programmer** [Cyfrin/foundry-full-course-cu#366] [Q&A]
I did everything right acc. to the video.....generated a contract then tried to verify this code on sepolia etherscan:

` // SPDX-License-Identifier: MIT
pragma solidity ^0.8.18; //Solidity Version

contract SimpleStorage {
    uint256 myFavNum;

    struct Person {
        uint256 favNum;
        string name;
    }

    Person[] public listOfPeople;

    mapping(string => uint256) public nameToNum;

    function store(uint256 _favNum) public virtual {
        myFavNum = _favNum;
    }

    function retrieve() public view returns (uint256) {
        return myFavNum;
    }

    function addPerson(string memory _name, uint256 _favNum) public {
        listOfPeople.push(Person(_favNum, _name));
        nameToNum[_name] = _favNum;
    }
}
`

I tried to select 0.8.18, 0.8.19, 0.8.20 all three compilers one by one on etherscan but it gave error.
I even tried to change the code a bit by doing this `0.8.18` instead of this `^0.8.18` then selecting 0.8.18 on etherscan but it still gives the same error.

@PatrickAlphaC  said you can run into errors while verifying manually. So should I keep trying or just move on??

**Comment by pyromaus** [comment]
IMO i've never successfully verified a contract on etherscan manually using their UI.. maybe once.
but you could try using the foundry terminal.

make sure you have a `.env` file in your project folder and the API key in it. sign up at etherscan to get an API key
```
ETHERSCAN_API_KEY=123abc789xyz247365
```
use the command to inject your `.env` into the foundry terminal
```
source .env
```
then this command will let foundry try to verify it 
```
forge verify-contract --chain-id 11155111 --etherscan-api-key $ETHERSCAN_API_KEY <your existing contract address you want to verify> src/SimpleStorage.sol:SimpleStorage --watch
```
if it doesn't work try running `foundryup` and try again

**Comment by Raghav-Programmer** [comment]
oh ok.....got it....thanks!!
